tensorflow/training/docker/2.11/py3/cu112/Dockerfile.gpu

```diff
@@ -0,0 +1,430 @@
+FROM nvidia/cuda:11.2.2-base-ubuntu20.04 AS base_image
+
+ENV DEBIAN_FRONTEND=noninteractive \
+    LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib"
+
+RUN apt-get update \
+ && apt-get upgrade -y \
+ && apt-get autoremove -y \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+FROM base_image AS common
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# TensorFlow major.minor version
+ENV TF_VERSION=2.11
+
+# prevent stopping by user interaction
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+ENV PYTHONIOENCODING=UTF-8
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+
+# Set environment variables for MKL
+# For more about MKL with TensorFlow see:
+# https://www.tensorflow.org/performance/performance_guide#tensorflow_with_intel%C2%AE_mkl_dnn
+ENV KMP_AFFINITY=granularity=fine,compact,1,0
+ENV KMP_BLOCKTIME=1
+ENV KMP_SETTINGS=0
+ENV RDMAV_FORK_SAFE=1
+
+ARG PYTHON=python3.9
+ARG PIP=pip3
+ARG PYTHON_VERSION=3.9.10
+
+ARG OPEN_MPI_PATH=/opt/amazon/openmpi
+ARG EFA_PATH=/opt/amazon/efa
+ARG EFA_VERSION=1.17.2
+ARG OMPI_VERSION=4.1.1
+ARG BRANCH_OFI=1.4.0-aws
+
+ARG CUDA=11.2
+ARG CUDA_DASH=11-2
+ARG CUDNN=8.2.4.15-1
+
+ARG NCCL_VERSION=2.13.4
+
+# To be passed to ec2 and sagemaker stages
+ENV PYTHON=${PYTHON}
+ENV PYTHON_VERSION=${PYTHON_VERSION}
+ENV PIP=${PIP}
+
+RUN apt-get update && apt-get install -y --no-install-recommends --allow-unauthenticated --allow-downgrades  --allow-change-held-packages \
+   ca-certificates \
+   cuda-command-line-tools-${CUDA_DASH} \
+   cuda-cudart-dev-${CUDA_DASH} \
+   libcufft-dev-${CUDA_DASH} \
+   libcurand-dev-${CUDA_DASH} \
+   libcusolver-dev-${CUDA_DASH} \
+   libcusparse-dev-${CUDA_DASH} \
+   curl \
+   emacs \
+   libcudnn8=${CUDNN}+cuda11.4 \
+   libgomp1 \
+   libfreetype6-dev \
+   libhdf5-serial-dev \
+   liblzma-dev \
+   libpng-dev \
+   libtemplate-perl \
+   libzmq3-dev \
+   hwloc \
+   git \
+   unzip \
+   wget \
+   libtool \
+   vim \
+   libssl1.1 \
+   openssl \
+   build-essential \
+   openssh-client \
+   openssh-server \
+   zlib1g-dev \
+   # Install dependent library for OpenCV
+   libgtk2.0-dev \
+   jq \
+ && apt-get update \
+ && apt-get install -y --no-install-recommends --allow-unauthenticated --allow-change-held-packages \
+   libcublas-dev-${CUDA_DASH} \
+   libcublas-${CUDA_DASH} \
+   # The 'apt-get install' of nvinfer-runtime-trt-repo-ubuntu1804-5.0.2-ga-cuda10.0
+   # adds a new list which contains libnvinfer library, so it needs another
+   # 'apt-get update' to retrieve that list before it can actually install the
+   # library.
+   # We don't install libnvinfer-dev since we don't need to build against TensorRT,
+   # and libnvinfer4 doesn't contain libnvinfer.a static library.
+   # nvinfer-runtime-trt-repo doesn't have a 1804-cuda10.1 version yet. see:
+   # https://developer.download.nvidia.cn/compute/machine-learning/repos/ubuntu1804/x86_64/
```

Comment:
Is this still applicable after having moved to TF 2.11 and CUDA 11.2?

Reply:
Is this still needed for TF2.11? If this is for CUDA 10?